### ISSUE TITLE
Environment variables for new tab are not updated

### ISSUE THREAD

**Issue by ocitrev** [microsoft/terminal#6434]
# Environment
```none
Windows build number: Version 10.0.19041.264
Windows Terminal version (if applicable): 1.0.1401.0
```

# Steps to reproduce

1. Open Windows Terminal
1. Add new environment variable `TEST=1` using **SystemPropertiesAdvanced.exe**
1. Check environment variable is first tab with `$env:TEST` or `echo %TEST%`
*As expected: current tab did not inherit new environment variables*
1. Create new Tab
1. Check environment variable in second tab with `$env:TEST` or `echo %TEST%`

# Expected behavior
New tab should inherit new environment variables even when changed/added after first Terminal tab is created.

# Actual behavior
New tab did not inherit new `TEST` environment variable.

Environment variables were not inherited from the system, a new Windows Terminal must be launched to inherit new or modified environment variables.  

# Conclusion
I feel this is a bug, When I change an environment variable I don't have to restart Explorer before starting a new application.  I think it should be the same with Terminal creating new tabs.

**Comment by zadjii-msft** [comment]
Thanks for the suggestion! Please make sure to [search for duplicates](https://github.com/microsoft/terminal/issues?q=is%3Aissue+%22Environment+variables%22+is%3Aclosed) when filing new issues. This is already being tracked by another issue on our repo - please refer to #1125 for more discussion.

/dup #1125
